### ISSUE TITLE
Cannot select sniper runtime in "Compatibility" tab of the game properties, on the Steam client

### ISSUE THREAD

**Issue by touraill-adobe** [ValveSoftware/steam-runtime#728]
I have all three runtimes installed, `scout`, `soldier`, `sniper`. In the properties of games, I can only force the use of the `scout` runtime or the `legacy` runtime, except some times (not sure when).

A few weeks ago I already noticed this behavior, but usually I would

1. force the use of the `scout` runtime
2. launch the app => it failed to launch
3. go back to the compatibility settings
=> `sniper` was now available

It doesn't seem to be the case anymore. I can get `sniper` to be listed after launching a game for which `sniper` is specified though a mapping on Steamworks, but not for other games.

I need to be able to select `sniper` to see if my apps launch correctly in it, before creating mappings on SteamWorks, and also to test local builds (through "Add a non-Steam game to my library").

Could you explain how does the compatibility tab decides which runtime to expose to users, and how I can enable `sniper` for all games in my library, regardless of the Steamworks settings (if any)?

**Comment by kisak-valve** [comment]
Hello @touraill-adobe, please see https://github.com/ValveSoftware/steam-for-linux/issues/7430#issuecomment-1227168019. The short version is that you can't switch your entire game library over to Steam Linux Runtime 3.0 (Sniper) without breakage. If a game is configured on the game dev's side to use Steam Linux Runtime 3.0 (Sniper), then Steam should not let it use Steam Linux Runtime 1.0 (Scout) (see https://github.com/ValveSoftware/steam-for-linux/issues/9844).

If you're able to reliably make a game start with an unintended runtime, then that's something a Steam dev should look into.

**Comment by touraill-adobe** [comment]
Thanks for the link. I think I misunderstood the role of the "force the use of a specific compatibility tool": I thought it was a way to override the default choice made by the devs on the server, but in fact it is just a way to select _among the choices allowed by the devs_, is that correct?

>  you might be able to make a beta branch and configure that to use the runtime you want separately from the main game release.

I did that to make some specific tests but it is quite cumbersome to do that for all my games (all the games we published, I mean), and it doesn't work for local builds that are not uploaded to Steam. I'll reread a previous discussion I had with smcv where I think he mentioned some techniques to run commands in the runtime.

I'm closing this issue as I now understand that this is "as designed".
